src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/Android/FormsVideoView.android.cs

```diff
@@ -52,10 +52,14 @@ protected void ExtractMetadata(MediaMetadataRetriever retriever)
 
 		public override async void SetVideoURI(global::Android.Net.Uri? uri, IDictionary<string, string>? headers)
 		{
+			base.SetVideoURI(uri, headers);
```

Comment:
if we cache the `url` in a local variable, does it work?

```csharp
var localUri = uri;
// rest of the code use `localUri`
```

Reply:
It should work without any issues, but can you elaborate why we should do that? The call to the base method throws an exception (if the object is disposed), so it wouldn't matter if we cache the argument.

Reply:
That will avoid the usage of a disposed member. I looked into the SetMetadata method and can't see any dispose logical there... So it's something else that's disposing the URL, probably because we're using an async void and it's not waited.

Reply:
Take a look [here](https://github.com/xamarin/XamarinCommunityToolkit/blob/22b1d4ccf45555c2ed89ecb94e9ff93a5cf61ab3/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/Android/MediaElementRenderer.android.cs#L339). The uri that is passed down to `SetVideoURI()` is never disposed. To elaborate, MediaElement is working correctly unless you close the parent page while it's loading metadata, which will `Dispose()` the renderer and in turn dispose `FormsVideoView`. In the old implementation, metadata awaiting was before the base method call. If the user closes the page while we are awaiting metadata updates, then the next call to base.SetVideoURI will throw and since we are in a async void method we can't catch any exceptions outside. Of course, we can do this after awaiting:

```suggestion
try
{
  base.SetVideoURI(uri, headers);
}
catch(ObjectDisposedException) { }
```
but that is bad code in my opinion.

I caught this bug by accident, by opening and closing the page really fast. After rearranging them, there were no more exceptions. The other bug (not set to null after disposing) was more frequent since the `MetadataRetrieved` event was raised before the base call and the renderer tried to access the disposed instance first.

Reply:
@anvuks I believe that your PR fixes the issue. I just want to make sure that your PR doesn't cause any side effects. I'll test it tomorrow and merge it if everything works during my test. Also, I invite @peterfoot to take a look on this if he can

Reply:
@pictos  @anvuks  can you guys tell me where can i find the file that need to be edited by adding try and catch ???